homeassistant/components/lidarr/sensor.py

```diff
@@ -0,0 +1,162 @@
+"""Support for Lidarr."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from copy import deepcopy
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Generic
+
+from aiopyarr import LidarrQueueItem, LidarrRootFolder
+
+from homeassistant.components.sensor import (
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import DATA_GIGABYTES
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from . import LidarrEntity
+from .const import BYTE_SIZES, DOMAIN
+from .coordinator import LidarrDataUpdateCoordinator, T
+
+
+def get_space(data: list[LidarrRootFolder], name: str) -> str:
+    """Get space."""
+    space = []
+    for mount in data:
+        if name in mount.path:
+            mount.freeSpace = mount.freeSpace if mount.accessible else 0
+            space.append(mount.freeSpace / 1024 ** BYTE_SIZES.index(DATA_GIGABYTES))
+    return f"{space[0]:.2f}"
+
+
+def get_modified_description(
+    description: LidarrSensorEntityDescription, mount: LidarrRootFolder
+) -> tuple[LidarrSensorEntityDescription, str]:
+    """Return modified description and folder name."""
+    desc = deepcopy(description)
+    name = mount.path.rsplit("/")[-1].rsplit("\\")[-1]
+    desc.key = f"{description.key}_{name}"
+    desc.name = f"{description.name} {name}".capitalize()
+    return desc, name
+
+
+@dataclass
+class LidarrSensorEntityDescriptionMixIn(Generic[T]):
+    """Mixin for required keys."""
+
+    value_fn: Callable[[T, str], str]
+
+
+@dataclass
+class LidarrSensorEntityDescription(
+    SensorEntityDescription, LidarrSensorEntityDescriptionMixIn, Generic[T]
+):
+    """Class to describe a Lidarr sensor."""
+
+    attributes_fn: Callable[
+        [T], dict[str, StateType | datetime] | None
+    ] = lambda _: None
+    description_fn: Callable[
+        [LidarrSensorEntityDescription, LidarrRootFolder],
+        tuple[LidarrSensorEntityDescription, str] | None,
+    ] = lambda _, __: None
+
+
+SENSOR_TYPES: dict[str, LidarrSensorEntityDescription] = {
+    "disk_space": LidarrSensorEntityDescription(
+        key="disk_space",
+        name="Disk space",
+        native_unit_of_measurement=DATA_GIGABYTES,
+        icon="mdi:harddisk",
+        value_fn=get_space,
+        state_class=SensorStateClass.TOTAL,
+        description_fn=get_modified_description,
+    ),
+    "queue": LidarrSensorEntityDescription(
+        key="queue",
+        name="Queue",
+        native_unit_of_measurement="Albums",
+        icon="mdi:download",
+        value_fn=lambda data, _: data.totalRecords,
+        state_class=SensorStateClass.TOTAL,
+        attributes_fn=lambda data: {i.title: queue_str(i) for i in data.records},
```

Comment:
We want the attribute keys to be explicitly defined in the integration. We want to see the formatting and what they are. Please change this.